src/http/request.cr

```diff
@@ -121,6 +121,91 @@ class HTTP::Request
     BadRequest.new
   end
 
+  # :nodoc:
+  METHODS = %w(GET HEAD POST PUT DELETE CONNECT OPTIONS PATCH)
```

Comment:
I think they can be public, HTTP request methods are used in macros in web frameworks like [Kemal](https://github.com/kemalcr/kemal/blob/v0.25.2/src/kemal/dsl.cr#L9) and [Amber](https://github.com/amberframework/amber/blob/v0.29.0/src/amber/dsl/router.cr), and in [HTTP::Client](https://github.com/crystal-lang/crystal/blob/0.29.0/src/http/client.cr#L370).

The `TRACE` method is missing.
The RFCs can also be put in the comment description:
`PATCH`:  https://tools.ietf.org/html/rfc5789#section-2
other methods: https://tools.ietf.org/html/rfc7231#section-4.

Can be changed to a Tuple, or maybe an enum.

Reply:
Thanks, I added `TRACE`.

There's no need to cover all cases here, if a method is missing it will still work though a string will be allocated for that method.

Reply:
How about making them public, as in [Golang](https://golang.org/pkg/net/http/#pkg-constants)?

Reply:
This PR isn't about adding new API. I'd rather not add API that I'm not sure how would one use.

Reply:
Ok so, better make it private then.